python-sdk/src/astro/files/operators/files.py

```diff
@@ -1,12 +1,15 @@
-from typing import Any
+from typing import TYPE_CHECKING, Any
 
 from airflow.decorators.base import get_unique_task_id
 from airflow.models import BaseOperator
-from airflow.utils.context import Context
 
 from astro.files.base import File
 from astro.files.locations import create_file_location
 
+if TYPE_CHECKING:
```

Comment:
same as https://github.com/astronomer/astronomer-providers/pull/619#discussion_r962155341

thoughts?

Reply:
@josh-fell & @kaxil I like @kaxil 's proposal of having a compatibility module.
It would also be nice to see tests covering this issue - since we are using nox, it should be straightforward to add a test.

Reply:
Actually, any preference on having a pre-commit hook to check that the typing compat module is used vs. a unit test?

Reply:
Either is fine :)